### DISCUSSION TITLE
Autocreate accounts for first-time SSO logins

### DISCUSSION THREAD

**Discussion by vacri** [directus/directus#5625] [Archive]
So I'm setting up a new setup, and I want to use SSO so that my colleagues don't have to manage Yet Another Set Of Credentials. I ran into a problem where SSO doesn't work unless you already have an account created for you ('INVALID_USER'). This means I'll still need an admin to create a user and also apply a role to that user before SSO will work for them.

An optional feature that autocreated accounts for SSO users would be great, and in its basic form I think would only need a couple of parameters to work:

- allow only autocreation of users from a whitelist of domains
- a specific role to assign by default to autocreated users

This way a user can use their Google Workspace (in our case) account to just simply log in and get started, without an admin having to 'prepare the way' first. The system will autocreate a user and give them the assigned role.

**Comment by benhaynes** [comment]
Duplicate: https://github.com/directus/directus/discussions/3956

**Comment by younky-yang** [comment]
Even I specified the ADMIN_USER and ADMIN_PASSWORD that same as my SSO info, it still gives me INVALID_USER error. I tried also --skipAdminInit, but still no luck. 

Basically the SSO doesn't work for me.
